vulcanai/net.py

```diff
@@ -114,44 +87,113 @@ def __init__(self, name, dimensions, input_var, y, config,
+
+        Returns:
+            train_step: The Op for training.
+        """
+        print("Creating {} Trainer...".format(self.name))
+        # Add a scalar summary for the snapshot loss.
+        tf.summary.scalar('loss', self.cost)
+        # calculate updates using ADAM optimization gradient descent
+        if self.optimizer == 'adam':
+            optimizer = optimizers[self.optimizer](
+                            learning_rate=learning_rate,
+                            beta1=0.9,
+                            beta2=0.999,
+                            epsilon=1e-08
+            )
+        elif self.optimizer == 'sgd':
+            optimizer = optimizers[self.optimizer](
+                            learning_rate=learning_rate
+            )
+        else:
+            updates = None
+            ValueError("No optimizer found")
+
+        with tf.name_scope("train"):
+            # Variable to track the global step.
+            global_step = tf.Variable(0, name='global_step', trainable=False)
+            train_step = optimizer.minimize(self.cost, global_step=global_step)
+
+        return train_step
 
-        returns a network.
+    def evaluation(self, logits, labels_placeholder):
         """
+        Records the accuracy of the network.
+
+        Returns: A scalar float32 tensor with the number of examples
+                 that were predicted correctly.
+        """
+        print("Creating {} Validator...".format(self.name))
+
+        with tf.name_scope("accuracy"):
+            # check how much error in prediction
+            if self.val_cost is None:
+                if self.num_classes is None or self.num_classes == 0:
+                    self.val_cost = self.mse_loss(logits, labels_placeholder)
+                    val_acc = tf.constant(0)
+                else:
+                    self.val_cost = self.cross_entropy_loss(logits, labels_placeholder)
+                    # check the accuracy of the prediction
+                    if self.num_classes > 1:
+                        val_acc = tf.reduce_mean(tf.cast(tf.equal(
+                                              tf.argmax(logits, axis=1),
+                                              tf.argmax(labels_placeholder, axis=1)),
+                                              tf.float32))
+                    elif self.num_classes == 1:
+                        val_acc = tf.reduce_mean(tf.cast(tf.equal(
+                                              tf.round(logits), labels_placeholder),
+                                              tf.float32))
+            tf.summary.scalar("accuracy", val_acc)
+
+        return val_acc
+
+    def create_network(self, config, nonlinearity):
+
         import jsonschema
         import schemas
         mode = config.get('mode')
         if mode == 'dense':
             jsonschema.validate(config, schemas.dense_network)
 
-            network = self.create_dense_network(
+            network = self.build_dense_network_tf(
```

Comment:
I think you probably don't want to append the _tf to the function names.

Reply:
Agreed^

Reply:
Yes I agree. I will remove it.